source/extensions/quic_listeners/quiche/platform/quic_text_utils_impl.h

```diff
@@ -0,0 +1,128 @@
+#pragma once
+
+#include "absl/strings/ascii.h"
+#include "absl/strings/escaping.h"
+#include "absl/strings/match.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/str_split.h"
+#include "quiche/quic/platform/api/quic_string.h"
+#include "quiche/quic/platform/api/quic_string_piece.h"
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+namespace quic {
+
+class QuicTextUtilsImpl {
+public:
+  static bool StartsWith(QuicStringPiece data, QuicStringPiece prefix) {
+    return absl::StartsWith(data, prefix);
+  }
+
+  static bool EndsWithIgnoreCase(QuicStringPiece data, QuicStringPiece suffix) {
+    return absl::EndsWithIgnoreCase(data, suffix);
+  }
+
+  static QuicString ToLower(QuicStringPiece data) { return absl::AsciiStrToLower(data); }
+
+  static void RemoveLeadingAndTrailingWhitespace(QuicStringPiece* data) {
```

Comment:
Can we use absl::StripAsciiWhitespace given that QuicStringPiece is string_view under the hood?

Reply:
Oh! Yes, definitely. Somehow I missed that method when looking for one to use, thanks for pointing it out.